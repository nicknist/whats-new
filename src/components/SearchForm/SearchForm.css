input, button {
  background: white;
  height: 50px;
}

input {
  width: 60vw;
}

button {
  color: black;
  font-weight: bold;
  border-radius: 10%;
  width: 6vw;
  align-self: center;
}

#search {
  padding: 0%;
}

form {
  display: flex;
}

.search-bar {
  margin-right: 0%;
}
