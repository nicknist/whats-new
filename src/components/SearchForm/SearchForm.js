import React from 'react';
import './SearchForm.css';

// SEARCHFORM COMPONENT CODE GOES HERE

const SearchForm = ({ searchArticle, changeInput, currentInput }) =>  {
  return (
    <header>
      <h1>Whats <span className="yellow">New?</span></h1>
      <form className="search-form">
        <label className="search-bar">
          <input onChange={changeInput} id="search" type="text" placeholder="Search for news article here"/>
        </label>
        <button onClick={() => searchArticle(currentInput)} type="button">Search</button>
      </form>
    </header>
  )
}

export default SearchForm;
