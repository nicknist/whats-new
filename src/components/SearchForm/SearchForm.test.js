import React from 'react';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  it('should make the search bar with a header tag', () => {
    const wrapper = shallow(<SearchForm searchArticle={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke the searchArticle function when clicked', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<SearchForm searchArticle={mockFunction} currentInput={''} changeInput={jest.fn()}/>);

    wrapper.find('button').simulate('click');

    expect(mockFunction).toHaveBeenCalled();
  })
})
