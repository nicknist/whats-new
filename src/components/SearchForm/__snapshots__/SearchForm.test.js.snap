// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should make the search bar with a header tag 1`] = `
<header>
  <h1>
    Whats 
    <span
      className="yellow"
    >
      New?
    </span>
  </h1>
  <form
    className="search-form"
  >
    <label
      className="search-bar"
    >
      <input
        id="search"
        placeholder="Search for news article here"
        type="text"
      />
    </label>
    <button
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </form>
</header>
`;
